client/store/users/AuthContext.jsx

```diff
@@ -56,12 +59,16 @@ export function AuthProvider({ children }) {
   }
 
   useEffect(() => {
-    const unsubscribe = auth.onAuthStateChanged((user) => {
+    auth.onAuthStateChanged(async (user) => {
       setCurrentUser(user);
+      if (user) {
+        const newToken = await user.getIdToken();
+        setToken(newToken);
+      } else {
+        setToken('');
+      }
```

Comment:
ahhh I was wondering where you were getting `auth.token` in api.js. I tried to do this very thing and couldn't get it to work.  I even used `onAuthStateChanged`.  The trick is `useEffect` apparently.